pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -419,6 +420,8 @@ protected ConsumerImpl(PulsarClientImpl client, String topic, ConsumerConfigurat
                 "The number of messages currently sitting in the consumer receive queue", topic, attrs);
         bytesPrefetchedGauge = ip.newUpDownCounter("pulsar.client.consumer.receive_queue.size", Unit.Bytes,
                 "The total size in bytes of messages currently sitting in the consumer receive queue", topic, attrs);
+        messageAvailablePermitsGauge = ip.newUpDownCounter("pulsar.client.consumer.available_permits.count",
```

Comment:
If you copied from above you need to notice `.count` was added since there are two sub-metrics for `receive_queue`. In yours you only have one. 
Also, thinking about available. Maybe if we know the limit:

`pulsar.client.consumer.permit.remaining`
`pulsar.client.consumer.permit.limit`

if we don't know the limit then maybe just ``pulsar.client.consumer.permits.remaining`

See naming guidelines here https://opentelemetry.io/docs/specs/semconv/general/metrics/

Reply:
> If you copied from above you need to notice `.count` was added since there are two sub-metrics for `receive_queue`. In yours you only have one.

I noticed that `.size` is the data of the message body, but `permits` is just a counter and does not have a size itself, so `.size` may not be needed.

---

> Also, thinking about available. Maybe if we know the limit:
> 
> `pulsar.client.consumer.permit.remaining` `pulsar.client.consumer.permit.limit`

`pulsar.client.consumer.permit.remaining`= getCurrentReceiverQueueSize() / 2- available

`pulsar.client.consumer.permit.limit`= getCurrentReceiverQueueSize() / 2

Do you mean this?

Reply:
Yes.

Also I didn't write that `size` is needed :) Only that we don't need `.count` and added suggestion how to name it.

Reply:
Thank you for your follow-up, the changes have been made.  

PTAL.